src/m365/spp/commands/model/model-remove.spec.ts

```diff
@@ -0,0 +1,248 @@
+  });
+
+
+  it('correctly handles an access denied error', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        throw {
+          error: {
+            "odata.error": {
+              message: {
+                lang: "en-US",
+                value: "Attempted to perform an unauthorized operation."
+              }
+            }
+          }
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { verbose: true, siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f', force: true } }),
+      new CommandError('Attempted to perform an unauthorized operation.'));
+  });
+
+
+  it('deletes model by id', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(true);
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        return {
+          WebTemplateConfiguration: 'CONTENTCTR#0'
+        };
+      }
+      throw 'Invalid request';
+    });
+
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f' } });
+    assert.strictEqual(stubDelete.lastCall.args[0].url, `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`);
+    assert(confirmationStub.calledOnce);
+  });
+
+  it('does not delete model when confirmation is not accepted', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(false);
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        return {
+          WebTemplateConfiguration: 'CONTENTCTR#0'
+        };
+      }
+      throw 'Invalid request';
+    });
+
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f' } });
+    assert(stubDelete.notCalled);
+    assert(confirmationStub.calledOnce);
```

Comment:
Instead of testing this in every test, let's just create a dedicated test where we check if the prompt is issued.